frontends/web/src/routes/device/bitbox02/setdevicename.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useRef, FunctionComponent } from 'react';
+import { useTranslation } from 'react-i18next';
+import { getDeviceInfo, setDeviceName } from '../../../api/bitbox02';
+import { Button, Input } from '../../../components/forms';
+import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
+import { alertUser } from '../../../components/alert/Alert';
+import { SettingsButton } from '../../../components/settingsButton/settingsButton';
+import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+
+type Props = {
+    deviceName?: string;
+    deviceID: string;
+}
+
+export const SetDeviceName: FunctionComponent<Props> = ({
+    deviceName,
+    deviceID,
+}) => {
+    const { t } = useTranslation();
+
+    const [active, setActive] = useState(false);
+    const [currentName, setCurrentName] = useState(deviceName);
+    const [name, setName] = useState('');
+    const [inProgress, setInProgress] = useState(false);
+
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const updateName = () => {
```

Comment:
this might be a good candidate for async/await syntax, but it's also fine as it is.

Reply:
I'll check, the only thing I don't like (or am not sure exactly) is for `.catch` to use `try {} catch(e) {} ` if the response fails and have to check the finally thing.

Reply:
IIRC there was an advantage over promises when using await/async, if I remember correctly better stacktraces.

I'll change to await/async ack? 

```diff
diff --git a/frontends/web/src/routes/device/bitbox02/setdevicename.tsx b/frontends/web/src/routes/device/bitbox02/setdevicename.tsx
index 8f47c4b4..da411182 100644
--- a/frontends/web/src/routes/device/bitbox02/setdevicename.tsx
+++ b/frontends/web/src/routes/device/bitbox02/setdevicename.tsx
@@ -42,16 +42,18 @@ export const SetDeviceName: FunctionComponent<Props> = ({
 
     const inputRef = useRef<HTMLInputElement>(null);
 
-    const updateName = () => {
+    const updateName = async () => {
         setInProgress(true);
-        setDeviceName(deviceID, name)
-            .then(() => getDeviceInfo(deviceID))
-            .then(({ name }) => setCurrentName(name))
-            .catch(() => alertUser('Device name could not be set'))
-            .finally(() => {
-                setActive(false);
-                setInProgress(false);
-            });
+        try {
+            await setDeviceName(deviceID, name);
+            const { name: newDeviceName } = await getDeviceInfo(deviceID);
+            setCurrentName(newDeviceName);
+        } catch (e) {
+            alertUser('Device name could not be set')
+        } finally {
+            setActive(false);
+            setInProgress(false);
+        }
     };
 
     return (
```

Reply:
lgtm